lonboard/geopandas.py

```diff
@@ -0,0 +1,296 @@
+from typing import Any, Dict, List, Optional, Union
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+
+from . import Map, basemap, viz
+from .colormap import apply_categorical_cmap, apply_continuous_cmap
+
+__all__ = ["LonboardAccessor"]
+
+
+@pd.api.extensions.register_dataframe_accessor("lb")
+class LonboardAccessor:
+    def __init__(self, pandas_obj):
+        self._validate(pandas_obj)
+        self._obj = pandas_obj
+
+    @staticmethod
+    def _validate(obj):
+        if not isinstance(obj, gpd.GeoDataFrame):
+            raise AttributeError("must be a geodataframe")
+
+    def explore(
+        self,
+        column: Optional[str] = None,
+        cmap: Optional[str] = None,
+        scheme: Optional[str] = None,
+        k: Optional[int] = 6,
+        categorical: bool = False,
+        elevation: Union[str, np.ndarray] = None,
+        extruded: bool = False,
+        elevation_scale: Optional[float] = 1,
+        alpha: Optional[float] = 1,
+        layer_kwargs: Optional[Dict[str, Any]] = None,
+        map_kwargs: Optional[Dict[str, Any]] = None,
+        classification_kwds: Optional[Dict[str, Any]] = None,
+        nan_color: Optional[Union[List[int], np.ndarray[int]]] = [255, 255, 255, 255],
+        color: Optional[str] = None,
+        vmin: Optional[float] = None,
+        vmax: Optional[float] = None,
+        wireframe: bool = False,
+        tiles: Optional[str] = None,
+        highlight: bool = False,
+        m: Optional[Map] = None,
+    ) -> Map:
+        """explore a dataframe using lonboard and deckgl
+
+        Keyword Args:
+            column : Name of column on dataframe to visualize on map.
+            cmap : Name of matplotlib colormap to use.
+            scheme : Name of a classification scheme defined by mapclassify.Classifier.
+            k : Number of classes to generate. Defaults to 6.
+            categorical : Whether the data should be treated as categorical or
+                continuous.
+            elevation : Name of column on the dataframe used to extrude each geometry or
+                an array-like in the same order as observations. Defaults to None.
+            extruded : Whether to extrude geometries using the z-dimension.
```

Comment:
I probably don't follow here. Shouldn't extruded be automatically True when elevation is provided? Like the two are tied together, no? Or can I use extruded without elevation? What does it extrude on, then, Z coordinate? How?

Reply:
i agree this is a little awkward. There's probably no need for the bool, though I was just trying to follow the viz/Layer API which treats them separately. Happy to go with consensus here.

Another small wrinkle is that sometimes in GIS or CAD software (IIRC anyway) 'elevation' sometimes refers to the height *beneath* the polygon, e.g. so you can drop the extruded poly onto a DEM for things like viewsheds. In that case I want to say this parameter is called 'extrusion_height' or something? No real opinion here, just something to consider in case folks are coming from that world

Reply:
i removed the bool so that its turned on automatically if elevation is passed